src/nerve_tools/configure_nerve.py

```diff
@@ -137,6 +143,31 @@ class ServiceInfo(TypedDict):
     deploy_group: Optional[str]
 
 
+# TODO(kaisen|2010-06-05: Use /listeners once github.com/envoyproxy/envoy/issues/6959 is resolved)
+def _get_envoy_config_dump(admin_port: int) -> dict:
+    try:
+        return requests.get(f'http://localhost:{admin_port}/config_dump').json()
+    except Exception as e:
+        logging.warning(f'Unable to get envoy config dump: {e}')
+        return {}
+
+
+def get_envoy_listeners(admin_port: int) -> dict:
+    envoy_listeners: Dict = {}
+    envoy_config_dump: Dict = _get_envoy_config_dump(admin_port)
+    for config in envoy_config_dump.get('configs', []):
+        for listener in config.get('dynamic_active_listeners', []):
+            listener_config = listener.get('listener', {})
+            if listener_config.get('name', '').endswith('ingress_listener'):
+                result = INGRESS_LISTENER_REGEX.match(listener_config['name'])
+                if result:
+                    service_name = result.group(1)
+                    local_host_port = result.group(2)
+                    envoy_listeners[f'{service_name}.{local_host_port}'] = \
+                        int(listener_config['address']['socket_address']['port_value'])
```

Comment:
this is kind of a far out edge case, but [technically](https://github.com/envoyproxy/data-plane-api/blob/cb050ce1e502dd7b75b10bf09c1ec81406db71e8/envoy/api/v2/core/address.proto#L107) the `address` may have either a `socket_address` field or a `pipe` field. And the `socket_address` field [may have](https://github.com/envoyproxy/data-plane-api/blob/cb050ce1e502dd7b75b10bf09c1ec81406db71e8/envoy/api/v2/core/address.proto#L48) either a `port_value` field or a `named_port` field. It's probably fine to crash on that case if we forget to deal with it.

Reply:
Made this a bit more resilient by passing over listeners that don't conform to the expected 'local_address' -> 'socket_address' -> 'port_value' hierarchy.